packages/react-components/tests/task-summary/task-summary-accordion.test.tsx

```diff
@@ -0,0 +1,143 @@
+  });
+
+  test('active style is applied ', () => {
+    task.state = RomiCore.TaskSummary.STATE_ACTIVE;
+    const root = render(<TaskSummaryAccordion tasks={[task]} />);
+    expect(root.getByText(task.task_id).parentElement?.className).toContain('makeStyles-active');
+  });
+
+  test('queue style is applied', () => {
+    task.state = RomiCore.TaskSummary.STATE_QUEUED;
+    const root = render(<TaskSummaryAccordion tasks={[task]} />);
+    expect(root.getByText(task.task_id).parentElement?.className).toContain('makeStyles-queued');
+  });
+
+  test('completed style is applied', () => {
+    task.state = RomiCore.TaskSummary.STATE_COMPLETED;
+    const root = render(<TaskSummaryAccordion tasks={[task]} />);
+    expect(root.getByText(task.task_id).parentElement?.className).toContain('makeStyles-completed');
+  });
+
+  test('failed style is applied', () => {
+    task.state = RomiCore.TaskSummary.STATE_FAILED;
+    const root = render(<TaskSummaryAccordion tasks={[task]} />);
+    expect(root.getByText(task.task_id).parentElement?.className).toContain('makeStyles-failed');
+  });
+});
+
+describe('Components gets the correct label on specifics states', () => {
+  let task: RomiCore.TaskSummary;
+  beforeEach(() => {
+    task = Object.values(getTaskObject())[0];
+  });
+
+  test('Shows ACTIVE label', () => {
+    task.state = RomiCore.TaskSummary.STATE_ACTIVE;
+    const root = render(<TaskSummaryAccordionInfo task={task} />);
+    expect(root.getByText('ACTIVE')).toBeTruthy();
+  });
+
+  test('Shows QUEUE label', () => {
+    task.state = RomiCore.TaskSummary.STATE_QUEUED;
+    const root = render(<TaskSummaryAccordionInfo task={task} />);
+    expect(root.getByText('QUEUED')).toBeTruthy();
+  });
+
+  test('Shows COMPLETED label', () => {
+    task.state = RomiCore.TaskSummary.STATE_COMPLETED;
+    const root = render(<TaskSummaryAccordionInfo task={task} />);
+    expect(root.getByText('COMPLETED')).toBeTruthy();
+  });
+
+  test('Shows FAILED label', () => {
+    task.state = RomiCore.TaskSummary.STATE_FAILED;
+    const root = render(<TaskSummaryAccordionInfo task={task} />);
+    expect(root.getByText('FAILED')).toBeTruthy();
+  });
+});
+
+test('Get name of the actor from status', () => {
+  const rawStatus = 'Finding a plan for [tinyRobot/tinyRobot1] to go to [23] | Remaining phases: 6';
+  const actor = getActorFromStatus(rawStatus);
+  expect(actor).toEqual(['[tinyRobot/tinyRobot1]']);
+});
+
+test('Sorts task array correctly', () => {
+  const tasks = sortTasks(getTaskObject());
+  expect(tasks[0].state).toBe(RomiCore.TaskSummary.STATE_ACTIVE);
+  expect(tasks[1].state).toBe(RomiCore.TaskSummary.STATE_QUEUED);
+});
+
+test('Sorts task by submission time correctly', () => {
+  const tasks = Object.values(getTaskObject());
+  const sortedTasks = sortTasksBySubmissionTime(tasks);
+  expect(sortedTasks[0].submission_time.nanosec).toBe(1519129858900);
+  expect(sortedTasks[1].submission_time.nanosec).toBe(1519129853500);
+});
```

Comment:
Should we add a test to sort by multiple values?

Task 1: Active, time 500
Task 2: Active, time 1000
Task 3: Queued, time 1500

Expected result, [ Task2, Task1, Task3 ];

Reply:
Added